### DISCUSSION TITLE
tmux trackpad scrolling, when connected over SSH, is janky and jittery on ghostty when compared to Terminal.app

### DISCUSSION THREAD

**Discussion by nighthawk469** [ghostty-org/ghostty#3025] [Feature Requests, Ideas]
tmux trackpad scrolling, when connected over SSH, is janky and jittery on ghostty when compared to Terminal.app
On a 2023 14" Macbook Pro.

tmux has mouse mode enabled
`set-option -g mouse on`

The remote computer I am connected to is Ubuntu 20.04.

It feels like Terminal.app is doing a better job at filtering the small trackpad movements and only reacts to larger trackpad movements, when compared to ghostty.

tmux running locally without ssh on MacOS does NOT have this jittering issue.

Video uploads are not working on this github issue but I do have screen recordings.

**Comment by mitchellh** [comment]
Converted to a discussion until more details are determined.

**Comment by nighthawk469** [comment]
https://github.com/user-attachments/assets/78607ce5-e3a4-46e6-b0cc-8c95c058ccd1


https://github.com/user-attachments/assets/dfdfe465-8713-4c9e-8856-abd15f1826dc

**Comment by nighthawk469** [comment]
I originally reported this with tmux 3.0 but after testing with tmux 3.5 the issue is gone. Seems to have been fixed somewhere.

**Comment by nighthawk469** [comment]
Reopening because this still happens severley with regular vim over ssh inside tmux

In fact this scrolling issue happens with ghostty using vim (8.1) inside ssh, with or without tmux.

**Reply by mitchellh** [reply]
We of course have hundreds if not thousands of people using this exact workflow and I haven’t heard this issue before. 

Let me ask the obvious first: since you’re over SSH, does the remote end have a valid terminfo and is it recognizing Ghostty properly? This seems like it could be one of those scenarios it’s not since the behaviors are weird in those cases. Figured I’d start with the easy case :)

**Reply by nighthawk469** [reply]
> does the remote end have a valid terminfo and is it recognizing Ghostty properly

Not sure. I am running the infocmp command you recommend, but my Ubuntu is kind of old (20.04) and it gives a warning. I'm not sure if this is causing a problem

```
infocmp -x | ssh office -- tic -x -
"<stdin>", line 2, col 31, terminal 'xterm-ghostty': older tic versions may treat the description field as an alias

tic -V
ncurses 6.2.20200212
```

I don't see a TERMINFO environment variable on my remote.

```
env|grep -iE "term|ghostty"
TERM=xterm-ghostty
```

**Reply by 00-kat** [reply]
Very late, but that warning is harmless.

**Comment by mtaras0** [comment]
Just wanted to confirm that I was having the same issue (jumpy scroll inside tmux over ssh) but it was resolved by upgrading tmux from 3.2a to 3.5a.

**Comment by Codesmith28** [comment]
wanted to confirm that this issue also happens on neovim as well, 
when i use neovim with ghostty - it jitters a lot and also performs hscroll which is not intended, 
however when i use something like iterm2 it scrolls perfectly fine - in one direction at a time...


ps: i am on macos

**Comment by manishie** [comment]
i just went from 3.2a to 3.6b and the problem went away
